Source/JavaScriptCore/runtime/Error.cpp

```diff
@@ -255,6 +255,33 @@ JSObject* addErrorInfo(VM& vm, JSObject* error, int line, const SourceCode& sour
     return error;
 }
 
+JSObject* createTypeErrorCopy(JSGlobalObject* globalObject, JSValue error)
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_CATCH_SCOPE(vm);
+    String errorString = "Error encountered during evaluation"_s;
+
+    if (error.isPrimitive())
+        errorString = error.toWTFString(globalObject);
+    else if (error.isObject()) {
+        auto structure = error.asCell()->structure();
+        if (!structure->isProxy()) {
+            auto slot = PropertySlot(error, PropertySlot::InternalMethodType::GetOwnProperty);
+            if (error.getOwnPropertySlot(globalObject, vm.propertyNames->message, slot)) {
+                scope.assertNoException();
+                if (slot.isValue()) {
+                    JSValue message = slot.getValue(globalObject, vm.propertyNames->message);
+                    scope.assertNoException();
```

Comment:
This is not correct when termination exception happens. So, use

```
RETURN_IF_EXCEPTION(scope, {})
```

Reply:
Done 👍